File: 1817-finding-the-users-active-minutes/1817-finding-the-users-active-minutes.py
class Solution:
    def findingUsersActiveMinutes(self, logs: List[List[int]], k: int) -> List[int]:
        d = {}
        res = [0 for i in range(k)]
        for l in logs:
            i, t = l[0], l[1]
            if i not in d.keys():
                d[i] = {t}
            else:
                d[i].add(t)
        for i in d.values():
            res[len(i)-1]+=1
        return res

File: 1817-finding-the-users-active-minutes/1817-finding-the-users-active-minutes.cpp
class Solution {
public:
    vector<int> findingUsersActiveMinutes(vector<vector<int>>& logs, int k) {
        unordered_map<int, unordered_set<int>> um;;
        vector<int> answer(k, 0);
        for (auto v : logs) {
            um[v[0]].insert(v[1]);
        }
        for (auto p : um) answer[p.second.size()-1]++;
        return answer;
    }
};